### PR TITLE
Custom metrics

### PR DESCRIPTION
**Pull request by Ayaro**
WIP

### CODE REVIEW

**Comment by mfeurer** [comment]
Could you please add comments/description to the newly added metrics classes?

**Comment by Ayaro** [comment]
@mfeurer Ready for review

File: autosklearn/constants.py [hunk 1]
```diff
@@ -22,6 +23,7 @@
      'multilabel.classification': MULTILABEL_CLASSIFICATION,
      'regression': REGRESSION}
 
+'''
```
Comment (mfeurer) [inline autosklearn/constants.py]: Can you please remove such code instead of commenting it?

File: autosklearn/metrics/classification_metrics.py [hunk 2]
```diff
@@ -9,20 +9,21 @@
 import scipy as sp
 import scipy.stats
 from autosklearn.constants import MULTICLASS_CLASSIFICATION, \
-    BINARY_CLASSIFICATION, METRIC_TO_STRING, MULTILABEL_CLASSIFICATION
+    BINARY_CLASSIFICATION, MULTILABEL_CLASSIFICATION
 from autosklearn.metrics.util import log_loss, prior_log_loss, \
     binarize_predictions, normalize_array, create_multiclass_solution
 
-
+'''
```
Comment (mfeurer) [inline autosklearn/metrics/classification_metrics.py]: Same as above.

File: autosklearn/metrics/factories.py [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+                               self.scoring_function,
+                               task_type)
+        else:
+            raise ValueError('Task type %s is not supported by the builder.'
+                             % task_type)
+
+    def get_scoring_function(self):
+        pass
+
+'''
```
Comment (mfeurer) [inline autosklearn/metrics/factories.py]: Can you move this into the metric builder and document this further?

File: autosklearn/metrics/regression_metrics.py [hunk 4]
```diff
@@ -6,9 +6,9 @@
 
 import numpy as np
 
-from autosklearn.constants import REGRESSION, METRIC_TO_STRING
-
+from autosklearn.constants import REGRESSION
 
+'''
```
Comment (mfeurer) [inline autosklearn/metrics/regression_metrics.py]: Same as above, please remove this code

File: autosklearn/metrics/factories.py [hunk 5]
```diff
@@ -0,0 +1,227 @@
...
+class AUCMetricBuilder(MetricBuilder):
+    metric_name = 'auc_metric'
+    scoring_function = staticmethod(auc_metric)
+    possible_metric_names = ['auc_metric', 'auc']
+    possible_task_types = [BINARY_CLASSIFICATION, MULTICLASS_CLASSIFICATION,
+                           MULTILABEL_CLASSIFICATION]
+
+
+class KnownMetricFactory():
+
```
Comment (mfeurer) [inline autosklearn/metrics/factories.py]: Can you please add a docstring to all metric factories?

File: autosklearn/metrics/factories.py [hunk 6]
```diff
@@ -0,0 +1,227 @@
+import six
+import importlib
+from autosklearn.constants import BINARY_CLASSIFICATION, \
+    MULTICLASS_CLASSIFICATION, MULTILABEL_CLASSIFICATION, REGRESSION
+from classification_metrics import acc_metric, bac_metric, pac_metric, \
+    f1_metric, auc_metric
+from regression_metrics import a_metric, r2_metric
+
+
+### Metric base class and implementations ###
```
Comment (mfeurer) [inline autosklearn/metrics/factories.py]: Can you please add a docstring for the module?

**Comment by mfeurer** [comment]
I just merged this into a different branch (refactor/evaluator) and github didn't realize. Closing this now.
